backend/benefit/applications/tests/test_applications_api.py

```diff
@@ -2035,3 +2048,52 @@ def test_application_status_last_changed_at(api_client, handling_application):
     assert response.data["status_last_changed_at"] == datetime(
         2021, 12, 1, tzinfo=pytz.UTC
     )
+
+
+def test_handler_application_default_ordering(handler_api_client):
+    f = faker.Faker()
+    combos = [
+        (ReceivedApplicationFactory, ApplicationStatus.RECEIVED),
+        (HandlingApplicationFactory, ApplicationStatus.HANDLING),
+        (DecidedApplicationFactory, ApplicationStatus.ACCEPTED),
+    ]
+    for _ in range(5):
+        for class_name, status in combos:
+            application = class_name()
+            random_datetime = f.past_datetime()
+            application.log_entries.filter(to_status=status).update(
+                created_at=random_datetime
+            )
+            Calculation.objects.filter(application__id=application.pk).update(
+                modified_at=random_datetime
+            )
+
+    def _expected_sort_key(obj):
+        handled_at_key = None
+        if (
+            log_entry := obj.log_entries.filter(
+                to_status__in=[
+                    ApplicationStatus.REJECTED,
+                    ApplicationStatus.ACCEPTED,
+                    ApplicationStatus.CANCELLED,
+                ]
+            )
+            .order_by("-created_at")
+            .first()
+        ):
+            handled_at_key = log_entry.created_at.timestamp()
+
+        return (
+            -handled_at_key if handled_at_key else float("-inf"),
```

Comment:
Why the float("-inf")? I tried this:
```python
>>> a=[(1, 2), (2, -54), (3, 0), (1, 0), (1, 4), (1, -5), (None, 2), (None, -3), (None, 4), (None, 0)]
>>> a
[(1, 2), (2, -54), (3, 0), (1, 0), (1, 4), (1, -5), (None, 2), (None, -3), (None, 4), (None, 0)]
>>> sorted(a)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '<' not supported between instances of 'NoneType' and 'int'
```

So sorting None and non-None types doesn't work normally. Ok.